component/upgrade/patch.sh

```diff
@@ -0,0 +1,19 @@
+#!/bin/bash
+set -e
+
+#Patch CRDS so that ArgoCD does not delete them during upgrade
+for crd in $CRDS_TO_PATCH; do
+	  exists=$(kubectl get crd "${crd}" --ignore-not-found)
+    if [ -z "$exists" ]; then
+      >&2 echo "WARNING: Skipping '${crd}': not found."
+      continue
+    fi
+    #Remove ArgoCD managed-by label from the CRD
+    kubectl label crd "${crd}" argocd.argoproj.io/instance-
+done
```

Comment:
Is labeling the right approach?
Why not add the prune annnotation? https://argo-cd.readthedocs.io/en/stable/user-guide/sync-options/#no-prune-resources

Reply:
we can use either, in any case the CRDs will not be managed by ArgoCD anymore in Crossplane 1.7.1 so we remove the annotation alltogether

Reply:
Added the annotation you suggested for providers and configuration manifests